src/hooks/useFhirClient.tsx

```diff
@@ -142,38 +174,49 @@ export function useFhirClient() {
         if (!accountHeaders || !activeProject?.id || !activeSubjectId) {
           throw new Error('Cannot mutate resource in current state');
         }
-        let resource = merge<ResourceType>({}, resourceToUpsert);
 
-        // Subject
-        resource.subject = {
-          reference: `Patient/${activeSubjectId}`,
-        };
+        const resource = toResource(resourceToUpsert);
 
-        // Project/dataset
-        if (
-          !resource.meta?.tag?.find(
-            (t) => t.system === 'http://lifeomic.com/fhir/dataset',
+        return httpClient
+          .post<ResourceType>(
+            `/v1/fhir/dstu3/${resource.resourceType}`,
+            resource,
+            {
+              headers: accountHeaders,
+            },
           )
-        ) {
-          resource = merge(resource, { meta: { tag: [] } });
-          resource.meta!.tag!.push({
-            system: 'http://lifeomic.com/fhir/dataset',
-            code: activeProject?.id,
-          });
+          .then((res) => res.data);
+      },
+    });
+  };
+
+  const useUpsertBundleMutation = () => {
+    return useMutation({
+      mutationFn: async (resources: ResourceType[]) => {
+        if (!accountHeaders || !activeProject?.id || !activeSubjectId) {
+          throw new Error('Cannot mutate resource in current state');
         }
 
-        // Date/time
-        if (resource.resourceType === 'Observation') {
-          const observation = resource as fhir3.Observation;
-          if (!observation.effectiveDateTime) {
-            observation.effectiveDateTime = formatISO(Date.now());
+        const bundle: Bundle<ResourceType> = {
+          resourceType: 'Bundle',
+          type: 'collection',
+          entry: resources.map((resource) => ({ resource })),
+        };
+
```

Comment:
For a future PR: verify that the resources count is less than 100, which is the max allowed by `patient-service`.